float32 data
float32 duration